internal/packaging/commands/snap.go

```diff
@@ -11,29 +19,43 @@ const (
 	snapProxySettingFormat   = `proxy.%s=%q`
 )
 
-// snapCmder is the packageCommander instantiation for snap-based systems.
-var snapCmder = packageCommander{
-	prereq:           makeNopCmd(),
-	update:           makeNopCmd(),
-	upgrade:          buildCommand(snapBinary, "refresh"),
-	install:          buildCommand(snapBinary, "install"),
-	remove:           buildCommand(snapBinary, "remove"),
-	purge:            buildCommand(snapBinary, "remove"),
-	search:           buildCommand(snapBinary, "info %s"),
-	isInstalled:      buildCommand(snapBinary, "list %s"),
-	listAvailable:    makeNopCmd(),
-	listInstalled:    buildCommand(snapBinary, "list"),
-	addRepository:    makeNopCmd(),
-	listRepositories: makeNopCmd(),
-	removeRepository: makeNopCmd(),
-	cleanup:          makeNopCmd(),
-	// Note: proxy.{http,https} available since snapd 2.28
-	getProxy:              buildCommand(snapBinary, "get system proxy"),
-	proxySettingsFormat:   snapProxySettingFormat,
-	noProxySettingsFormat: snapNoProxySettingFormat,
-	setProxy:              buildCommand(snapBinary, "set system %s"),
+// NewSnapPackageCommander returns a PackageCommander for snap-based systems.
+func NewSnapPackageCommander() SnapPackageCommander {
+	return SnapPackageCommander{}
+}
+
+// SnapPackageCommander provides runnable shell commands for snap-based package
+// operations.
+type SnapPackageCommander struct{}
+
+func (SnapPackageCommander) UpgradeCmd() string {
+	return buildCommand(snapBinary, "refresh")
 }
 
-func makeNopCmd() string {
-	return buildCommand(":", "#No action here")
+func (SnapPackageCommander) InstallCmd(packs ...string) string {
+	args := append([]string{snapBinary, "install"}, packs...)
+	return buildCommand(args...)
+}
+
+func (SnapPackageCommander) SetProxyCmds(settings proxy.Settings) ([]string, error) {
+	if settings.Ftp != "" {
+		return nil, errors.Errorf("ftp proxy not supported by snap")
+	}
+	if settings.NoProxy != "" {
+		return nil, errors.Errorf("NoProxy not supported by snap")
+	}
+	if settings.AutoNoProxy != "" {
+		return nil, errors.Errorf("AutoNoProxy not supported by snap")
+	}
```

Comment:
Thank-you.

These should be typed errors, that way we can trap them.

Reply:
Done